nodes/filter/nodes.go

```diff
@@ -0,0 +1,225 @@
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. Only Numeric ‘input’ values less than 'InMax' are passed to ‘output’. Output the last known input float value if the input is greater than 'OutMin'. Out put nil if any of the inputs are nil."))
+	return &OnlyLower{body, 0}, nil
+}
+
+func NewPreventDuplicates(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventDuplicates, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘input’ values which are equal to the previous ‘input’ value. Output the last known input float value if the subsequent input value is the same. Out put nil if the input is nil."))
+	return &PreventDuplicates{body, 0}, nil
+}
+
+// preventNull, output the last known value if the input became null.
+// lastValue is defaulted to be 0
+func (inst *PreventNull) Process() {
+	input, null := inst.ReadPinAsFloat(node.In)
+
+	if null {
+		inst.WritePinFloat(node.Out, inst.lastValue)
+	} else {
+		inst.WritePinFloat(node.Out, input)
+		inst.lastValue = input
+	}
+}
+
+// preventEqualFloat, output the last known value if two inputs equal to each other. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *PreventEqualFloat) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	match, matchNull := inst.ReadPinAsFloat(node.InputMatch)
+
+	if inputNull || matchNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if float.RoundTo(input, 2) != float.RoundTo(match, 2) {
+			inst.WritePinFloat(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePinFloat(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// preventEqualString, output the last known value if two inputs equal to each other. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be "defaultString"
+func (inst *PreventEqualString) Process() {
+	input, inputNull := inst.ReadPinAsString(node.In)
+	match, matchNull := inst.ReadPinAsString(node.InputMatch)
+
+	if inputNull || matchNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if input != match {
+			inst.WritePin(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePin(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// onlyBetween, Output the last known input float value if the input is not in range. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *OnlyBetween) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	min, minNull := inst.ReadPinAsFloat(node.InMin)
+	max, maxNull := inst.ReadPinAsFloat(node.InMax)
+
+	if inputNull || minNull || maxNull {
```

Comment:
if the `min` or `max` settings are `null` then make it function the same as the `only-lower` and `only-greater` nodes.  If `min` and `max` are both `null` just allow any value through.  Dont allow `null` output, just keep the last value.